packages/create-platformatic/src/index.mjs

```diff
@@ -81,11 +83,93 @@ async function importOrLocal ({ pkgManager, name, projectDir, pkg }) {
     await execa(pkgManager, ['install', pkg + version], { cwd: projectDir })
     spinner.succeed()
 
-    const fileToImport = resolve.sync(pkg, { basedir: projectDir })
+    const fileToImport = resolveModule.sync(pkg, { basedir: projectDir })
     return await import(pathToFileURL(fileToImport))
   }
 }
 
+async function findApplicationRoot (projectDir) {
+  if (existsSync(resolve(projectDir, 'package.json'))) {
+    return projectDir
+  }
+
+  const files = await glob('**/*.{js,mjs,cjs,ts,mts,cts}', { cwd: projectDir })
+
+  if (files.length > 0) {
+    return dirname(resolve(projectDir, files[0]))
+  }
+
+  return null
+}
+
+export async function determineApplicationType (projectDir) {
+  let rootPackageJson
+  try {
+    rootPackageJson = JSON.parse(await readFile(resolve(projectDir, 'package.json'), 'utf-8'))
+  } catch {
+    rootPackageJson = {}
+  }
+
+  const { dependencies, devDependencies } = rootPackageJson
+
+  if (dependencies?.next || devDependencies?.next) {
+    return ['@platformatic/next', 'Next.js']
+  } else if (dependencies?.['@remix-run/dev'] || devDependencies?.['@remix-run/dev']) {
+    return ['@platformatic/remix', 'Remix']
+  } else if (dependencies?.vite || devDependencies?.vite) {
+    return ['@platformatic/vite', 'Vite']
```

Comment:
are you sure this is if is correct? Given that astro depends on vite, I think astro should come first?

Reply:
Fixed!